### PR TITLE
Editing Chapter 1

### PR DESCRIPTION
**Pull request by steveklabnik**
I did some preliminary editing work with No Starch on the first chapter of the book, and here's some of the results. We're going to want to return to this later when @brson etc's new rustup work is done, so this is mostly just a first pass.

But, we agreed that having separate chapters for each of this bit of intro is a bit excessive. So let's move all of this intro stuff into one chapter.

I'd appreciate a careful review of this, as there was also some confusion about some things, which resulted in me taking one huge markdown file apart and splitting it back up, as well as some editor issues, so I _think_ this looks good, but double checking things matters!

/cc @aturon

### CODE REVIEW

File: src/doc/trpl/getting-started.md [hunk 1]
```diff
@@ -3,3 +3,612 @@
...
+the beginning of every Rust program. The first line says, "I’m declaring a
+function named `main` that currently takes no arguments and returns nothing."
+If there were arguments, they would go inside the parentheses (`(` and `)`),
+and because we aren’t returning anything from this function, we can omit the
+return type entirely.
+
+Also note that the function body is wrapped in curly braces (`{` and `}`). Rust
+requires these around all function bodies. It's considered good style to put
+the opening curly brace on the same line as the function declaration, with one
+line space in between.
```
Comment (apasel422) [inline src/doc/trpl/getting-started.md]: Should this just be "space" instead of "line space"?

File: src/doc/trpl/getting-started.md [hunk 2]
```diff
@@ -3,3 +3,612 @@
...
+This line does all of the work in this little program: it prints text to the
+screen. There are a number of details that are important here. The first is
+that it’s indented with four spaces, not tabs. If you configure your editor of
+choice to insert four spaces with the tab key, it will make your coding much
+more efficient.
+ 
+The second important part is the `println!()` line. This is calling a Rust
+*[macro]*, which is how metaprogramming is done in Rust. If it were calling a
+function instead, it would look like this: `println()` (without the !). We'll
+discuss Rust macros in more detail in Chapter XX, but for now you just need to
```
Comment (apasel422) [inline src/doc/trpl/getting-started.md]: I assume the "XX" here is meant to be filled in?

File: src/doc/trpl/getting-started.md [hunk 3]
```diff
@@ -3,3 +3,612 @@
...
+enter the following commands:
+
+```bash
+$ mkdir src
+$ mv main.rs src/main.rs
+$ rm main  # or 'del main.exe' on Windows
+```
+
+Cargo expects your source files to live inside a *src* directory, so do that
+first. This leaves the top level project directory (in this case,
```
Comment (apasel422) [inline src/doc/trpl/getting-started.md]: s/top level/top-level/

**Comment by steveklabnik** [comment]
Fixed up @apasel422 's comments, as well as fixing the build :sweat:

File: src/doc/trpl/getting-started.md [hunk 4]
```diff
@@ -3,3 +3,612 @@
...
+| `aarch64-apple-ios`           |  ✓  |     |     | ARM64 iOS                  |
+| `i686-unknown-freebsd`        |  ✓  |  ✓  |     | 32-bit FreeBSD             |
+| `x86_64-unknown-freebsd`      |  ✓  |  ✓  |     | 64-bit FreeBSD             |
+| `x86_64-unknown-openbsd`      |  ✓  |  ✓  |     | 64-bit OpenBSD             |
+| `x86_64-unknown-netbsd`       |  ✓  |  ✓  |     | 64-bit NetBSD              |
+| `x86_64-unknown-bitrig`       |  ✓  |  ✓  |     | 64-bit Bitrig              |
+| `x86_64-unknown-dragonfly`    |  ✓  |  ✓  |     | 64-bit DragonFlyBSD        |
+| `x86_64-rumprun-netbsd`       |  ✓  |     |     | 64-bit NetBSD Rump Kernel  |
+| `i686-pc-windows-msvc` (XP)   |  ✓  |     |     | Windows XP support         |
+| `x86_64-pc-windows-msvc` (XP) |  ✓  |     |     | Windows XP support         |
```
Comment (brson) [inline src/doc/trpl/getting-started.md]: This list is sure to be insta-outdated in dead-tree format...

Comment (steveklabnik) [inline src/doc/trpl/getting-started.md]: Yeah, I am not entirely sure what to do about that, exactly.

**Comment by brson** [comment]
r=me

**Comment by steveklabnik** [comment]
Thanks @brson! I didn't leave a comment for every change, but I should have addressed it all.

**Pull request merged**